hazelcast/src/main/java/com/hazelcast/instance/impl/DefaultNodeExtension.java

```diff
@@ -283,63 +288,85 @@ private void printNodeInfoInternal(BuildInfo buildInfo, String build) {
         systemLogger.fine("Configured Hazelcast Serialization version: " + buildInfo.getSerializationVersion());
     }
 
+    @SuppressWarnings("checkstyle:CyclomaticComplexity")
     private void printSecurityFeaturesInfo(Config config) {
-        StringBuilder sb = new StringBuilder("\n🔒Status of security related features:");
-        addFeatureInfo(sb, "Custom cluster name used", !Config.DEFAULT_CLUSTER_NAME.equals(config.getClusterName()));
+        StringBuilder sb = new StringBuilder("\n🔒Security recommendations and their status:");
+        addSecurityFeatureCheck(sb, "Use a custom cluster name", !Config.DEFAULT_CLUSTER_NAME.equals(config.getClusterName()));
         boolean isMulticastJoin = node.shouldUseMulticastJoiner(getActiveMemberNetworkConfig(config).getJoin());
```

Comment:
Seems this is reporting a false negative. Calling `node.shouldUseMulticastJoiner` after `discoveryService` initialization can fix this, now `discoveryService` is null at this step.

Reply:
Nice catch. :+1:

Reply:
Fixed by moving it to Node.